QuizApp Week 3/QuizApp3.js

```diff
@@ -0,0 +1,164 @@
+        });
+
+        questionDiv.appendChild(revealButton);
+        questionsList.appendChild(questionDiv);
+    });
+};
+
+// Search functionality
+document.getElementById('search-button').addEventListener('click', () => {
+    const searchTerm = document.getElementById('search-input').value.toLowerCase();
+    const filteredQuestions = quizQuestions.filter(quizQuestion =>
+        quizQuestion.question.toLowerCase().includes(searchTerm)
+    );
+    displayFilteredQuestions(filteredQuestions);
+});
+
+const displayFilteredQuestions = (filteredQuestions) => {
+    const questionsList = document.getElementById('questions-list');
+    questionsList.innerHTML = ''; // Clear existing list
+
+    filteredQuestions.forEach((quizQuestion, index) => {
+        const questionDiv = document.createElement('div');
+        questionDiv.classList.add('question-item');
+
+        const questionText = document.createElement('p');
+        questionText.textContent = `${index + 1}. ${quizQuestion.question}`;
+        questionDiv.appendChild(questionText);
+
+        const optionsList = document.createElement('ul');
+        quizQuestion.options.forEach(option => {
+            const optionItem = document.createElement('li');
+            optionItem.textContent = option.text;
+            optionsList.appendChild(optionItem);
+        });
+        questionDiv.appendChild(optionsList);
+
+        const revealButton = document.createElement('button');
+        revealButton.textContent = 'Reveal Correct Answer';
+        revealButton.addEventListener('click', () => {
+            const correctOptions = quizQuestion.options
+                .filter(option => option.isCorrect)
+                .map(option => option.text)
+                .join(', ');
+            alert(`Correct Answer(s): ${correctOptions}`);
+        });
+
+        questionDiv.appendChild(revealButton);
+        questionsList.appendChild(questionDiv);
+    });
+};
+
+// Add players and manage scores
+document.getElementById('start-quiz-button').addEventListener('click', () => {
+    const player1Name = document.getElementById('player1-name').value;
+    const player2Name = document.getElementById('player2-name').value;
+
+    if (!player1Name || !player2Name) {
+        alert('Please enter names for both players.');
+        return;
+    }
+
+    players = [
+        { name: player1Name, score: 0 },
+        { name: player2Name, score: 0 }
+    ];
+
+    updatePlayerScores();
+});
+
+const updatePlayerScores = () => {
+    const playerList = document.getElementById('player-list');
+    playerList.innerHTML = ''; // Clear current list
+
+    players.forEach((player, index) => {
+        const playerDiv = document.createElement('div');
+        playerDiv.classList.add('player-item');
+
+        const playerName = document.createElement('p');
+        playerName.textContent = `${player.name}: ${player.score} points`;
+        playerDiv.appendChild(playerName);
```

Comment:
I would prefer if you added the `playerName` `<p>` element into a different child `<div>` than the `<div>` you add the `correctButton` and `wrongButton` to.

Right now they are added in the same `<div>`, meaning the resulting HTML looks like:
```html
<div>
   <p>...</p>
   <button>...</button>
   <button>...</button>
</div>
```

This makes the <p> the same width as the buttons, and that makes the text look cramped. It should be easy to fix. 😉